core/LSU.cpp

```diff
@@ -111,13 +111,38 @@ namespace olympia
     // Receive new load/store instruction from Dispatch Unit
     void LSU::getInstsFromDispatch_(const InstPtr & inst_ptr)
     {
-        core_types::RegFile reg_file = core_types::RF_INTEGER;
-        const auto & srcs = inst_ptr->getSourceOpInfoList();
-        if(srcs.size() > 0){
-            reg_file = olympia::coreutils::determineRegisterFile(srcs[0]);
+        bool all_ready = true; // assume all ready
+        for(const auto & src_reg : inst_ptr->getRenameData().getSourceList())
```

Comment:
Yeah, this `for` loop would only check address operands, which are all INTEGER.  So all you need to do is
```
scoreboard_views_[core_types::RF_INTEGER]->isSet(inst_ptr->getSrcRegisterBitMask(core_types::RF_INTEGER);
```
and that's it.

Then, you check for data operands if (and only if) a store instruction.

Reply:
Addressed in [453b6c3](https://github.com/riscv-software-src/riscv-perf-model/pull/70/commits/453b6c3b19892012263c7985f4af0fc4717b05d1)